File: RepositoryPattern/SampleApplicationForGM/network-util/src/main/java/com/example/networkutil/NetworkSessionRequestInterceptor.java
package com.example.networkutil;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class NetworkSessionRequestInterceptor implements Interceptor {

    public NetworkSessionRequestInterceptor() {
    }

    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();

        builder.addHeader("Accept-Language", "en-US");
        Request request = builder.build();
        return chain.proceed(request);
    }

}

File: RepositoryPattern/SampleApplicationForGM/network-util/src/main/java/com/example/networkutil/NetworkSessionRequestInterceptor.kt
package com.example.networkutil

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class NetworkSessionRequestInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        builder.addHeader("Accept-Language", "en-US")
        val request = builder.build()
        return chain.proceed(request)
    }
}